packages/server/src/adapters/fetch/types.ts

```diff
@@ -1,5 +1,20 @@
-import type { AnyRouter, inferRouterContext } from '@trpc/core';
-import type { HTTPBaseHandlerOptions, TRPCRequestInfo } from '../../http';
+/**
+ * If you're making an adapter for tRPC and looking at this file for reference, you should import types and functions from `@trpc/server` and `@trpc/server/http`
+ *
+ * Do **not** import from `@trpc/core`
+ * @example
+ * ```ts
+ * import type { AnyTRPCRouter } from '@trpc/server'
+ * import type { HTTPBaseHandlerOptions } from '@trpc/server/http'
+ * ```
+ */
+// @trpc/server
+import type { AnyRouter, inferRouterContext } from '../../@trpc-server';
```

Comment:
if this should serve as examples for other adapters maybe import the smurfed, non-deprecated, AnyTRPCRouter ?

Reply:
Yeah, I just wanted to keep it unchanged for now to ensure nothing broke